app/api/client_groups.py

```diff
@@ -31,3 +88,53 @@ def get_group_id(self: TwingateClientProtocol, group_name: str) -> str | None:
         except (TransportQueryError, IndexError, KeyError):
             logging.exception("Failed to get resource")
             return None
+
+    def group_create(
+        self: TwingateClientProtocol,
+        group: GroupSpec,
+        user_ids: list[str] | None = None,
+    ) -> str:
+        user_ids = user_ids or []
+        result = self.execute_mutation(
+            "groupCreate",
+            MUT_CREATE_GROUP,
+            variable_values={
+                "name": group.name,
+                "userIds": user_ids,
+            },
+        )
+        return result["entity"]["id"]
+
+    def group_update(
+        self: TwingateClientProtocol,
+        group: GroupSpec,
+        user_ids: list[str] | None = None,
+    ) -> str:
+        user_ids = user_ids or []
+        result = self.execute_mutation(
+            "groupUpdate",
+            MUT_UPDATE_GROUP,
+            variable_values={
+                "id": group.id,
+                "name": group.name,
+                "userIds": user_ids,
+            },
+        )
+        return result["entity"]["id"]
+
+    def group_delete(self: TwingateClientProtocol, group_id: str):
+        try:
+            result = self.execute_mutation(
+                "groupDelete",
+                MUT_DELETE_GROUP,
+                variable_values={"id": group_id},
+            )
+
+            return bool(result["ok"])
+        except GraphQLMutationError as gql_err:
+            if "does not exist" in gql_err.error:
+                return True
```

Comment:
I think we don't need this logic anymore. We update the Public API to consistently return success (with `ok=True`) if an entity doesn't exist for all kinds of entity in https://github.com/Twingate/controller/pull/8735